### ISSUE TITLE
Compilation error when targeting iOS

### ISSUE THREAD

**Issue by OtaK** [kornelski/rust-security-framework#175]
Hi!

It seems some recent additions are breaking code targeting iOS: 

```
error[E0004]: non-exhaustive patterns: type `&item::Location` is non-empty
   --> /home/otak/.cargo/registry/src/github.com-1ecc6299db9ec823/security-framework-2.8.0/src/item.rs:528:19
    |
528 |             match location{
    |                   ^^^^^^^^
    |
note: `item::Location` defined here
   --> /home/otak/.cargo/registry/src/github.com-1ecc6299db9ec823/security-framework-2.8.0/src/item.rs:597:10
    |
597 | pub enum Location {
    |          ^^^^^^^^
    = note: the matched value is of type `&item::Location`
    = note: references are always considered inhabited
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern as shown
    |
528 ~             match location {
529 +                 _ => todo!(),
530 +             }
    |
```

It seems the `match` statement doesn't cover `DataProtectionKeychain` when targeting iOS

Cheers!

**Comment by kcking** [comment]
I believe this is handled [here](https://github.com/kornelski/rust-security-framework/blob/2274b8ca2ce1babd8a8dc2d704c86af866370c95/security-framework/src/item.rs#L528-L529). Can you post the exact build command you're using, as well as the exact version of `security-framework` in your `Cargo.lock` file?

**Comment by kornelski** [comment]
I can't reproduce it when compiling for iOS, but I was able to reproduce it when compiling the code for **Linux**, which makes this enum completely empty (since it's neither macos nor ios, so nothing to match).

**Comment by kornelski** [comment]
Fixed in 2.8.1
